File: app/src/main/java/com/wenitech/cashdaily/viewModel/HomeFragmentViewModel.java
package com.wenitech.cashdaily.viewModel;

import androidx.lifecycle.ViewModel;

import com.google.firebase.auth.FirebaseUser;
import com.wenitech.cashdaily.Data.LiveData.CajaLiveData;
import com.wenitech.cashdaily.Data.repository.FirebaseRepository;

public class HomeFragmentViewModel extends ViewModel {

    public CajaLiveData cajaLiveData = null;

    private FirebaseRepository firebaseRepository;

    public HomeFragmentViewModel() {
        this.firebaseRepository = FirebaseRepository.getInstance();
    }

    public void getUserBox(FirebaseUser user) {
        cajaLiveData = firebaseRepository.getUserBoxRepository(user);
    }
}

File: app/src/main/java/com/wenitech/cashdaily/features/home/viewModel/HomeFragmentViewModel.kt
package com.wenitech.cashdaily.features.home.viewModel

import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.*
import com.wenitech.cashdaily.data.model.Caja
import com.wenitech.cashdaily.data.model.User
import com.wenitech.cashdaily.domain.interaction.caja.GetUserAppBoxUseCase
import com.wenitech.cashdaily.domain.interaction.auth.GetProfileUserAppUseCase

class HomeFragmentViewModel @ViewModelInject constructor(
        private val getUserAppBoxUseCase: GetUserAppBoxUseCase,
        private val getProfileUserAppUseCase: GetProfileUserAppUseCase
) : ViewModel() {

    var _profileUser: MutableLiveData<User> = MutableLiveData()
    val profileUser: LiveData<User>
        get() = _profileUser

    init {
        getProfileUserApp()
    }

    val cajaLiveData = getUserAppBoxUseCase.execute().asLiveData()

    fun getProfileUserApp() {
        /*getProfileUserAppUseCase.execute().observeForever(Observer {
            _profileUserApp.value = it
        })*/
    }

}